Added a stiffened panel buckling model (#2600)
openturns/openturns: python/doc/usecases/use_case_stiffened_panel.rst

.. _use-case-stiffened-panel:

Stiffened panel buckling
========================

Introduction
------------

The following figure presents a stiffed panel subject to buckling on a military aircraft.

> This use-case implements a simplified model of buckling for a stiffened panel, detailed in [ko1994]_.

Reviewer: ```suggestion
This use-case implements a simplified model of buckling for a stiffened panel (see [ko1994]_).
```

.. figure:: ../_static/stiffened_panel_illustration.jpg
    :align: center
    :alt: buckling illustration
    :width: 100%

    **Figure 1.** Buckling of a stiffened panel.

.. figure:: ../_static/stiffened_panel_simulation.png
    :align: center
    :alt: buckling simulation
    :width: 100%

    **Figure 2.** 3D simulation of buckling.

.. figure:: ../_static/stiffened_panel_description.png
    :align: center
    :alt: stiffened panel geometry
    :width: 100%

    **Figure 3.** Parameterization of the stiffened panel.

This test case is composed of nine random variables:

> - :math:`E\sim\mathcal{TN}(110.0e9, 55.0e9, 99.0e9, 121.0e9)` : Young modulus (Pa)

Reviewer: Using $10e9$ has no particular mathematical meaning because there is no variable $e$ involved here. The mathematical value is $10^9$, of course.
```suggestion
- :math:`E\sim\mathcal{TN}(110.0 \times 10^{9}, 55.0e9, 99.0e9, 121.0e9)` : Young modulus (Pa)
```
Please fix the other ones similarily.

Author: I imitated the way the excellent Fire Satellite use case was written. I will fix it accordingly.

Author: I used the siunitx package to render the numerical values and the units, IMO it should be the way to go now.

- :math:`nu\sim\mathcal{U}(0.3675, 0.3825)` : Poisson coefficient (-)

- :math:`h_c\sim\mathcal{U}(0.0285, 0.0315)` : Distance between the mean surface of the hat and the foot of the Stiffener (m)

- :math:`\ell\sim\mathcal{U}(0.04655, 0.05145)` : Length of the stiffener flank (m)

- :math:`f_1\sim\mathcal{U}(0.0266, 0.0294)` : Width of the stiffener foot (m)

- :math:`f_2\sim\mathcal{U}(0.00627, 0.00693)` : Width of the stiffener hat (m)

- :math:`t\sim\mathcal{U}(8.02e-5, 8.181e-5)` : Thickness of the panel and the stiffener (m)

- :math:`a\sim\mathcal{U}(0.6039, 0.6161)` : Width of the panel (m)

- :math:`b_0\sim\mathcal{U}(0.04455, 0.04545)` : Distance between two stiffeners (m)

- :math:`p\sim\mathcal{U}(0.03762, 0.03838)` : Half-width of the stiffener (m)

The output of interest is:

- :math:`(N_{xy})_{cr}`: the critical shear force (N)

We assume that the input variables are independent except the :math:`f_1` and
:math:`f_2` for which we measure a Spearman correlation of :math:`\rho^S_{12}=-0.8`,
modelled using a :class:`~openturns.NormalCopula`.

> The critical load :math:`(\tau_{xy})_{cr}` of a stiffened panel subject to shear load is given by

Reviewer: ```suggestion
The critical load :math:`(\tau_{xy})_{cr}` of a stiffened panel subject to shear load is:
```

.. math::
  (\tau_{xy})_{cr}=k_{xy}\frac{\pi^2D}{b_0^2t_s}

> where

Reviewer: ```suggestion
where:
```

- :math:`a` is the width of the panel;

- :math:`b_0` is the width between too consecutive stiffener feet;

- :math:`t_s` is the thickness of the panel main surface;

- :math:`E_s` is the Young modulus of the panel main surface;

- :math:`\nu_s` is the Poisson coefficient of the panel main surface;

> - :math:`D=\frac{E_st_s^3}{12(1-\nu_s^2)}` is the bending coefficient of the
>   panel main surface;

Reviewer: The equation deserves to be centered, because it is a part of the model and is difficult to read otherwise:
```suggestion
- :math:`D` is the bending coefficient of the
  panel main surface:
.. math:
    D = \frac{E_st_s^3}{12(1-\nu_s^2)};
```

- :math:`k_{xy}` is the load factor associated to shear buckling. It is given as

>   a function of :math:`\frac{b_0}{a}` through the empirical relation

Reviewer: ```suggestion
  a function of :math:`\frac{b_0}{a}` through the empirical equation
```

>   :math:`k_{xy}=5.35 + 4\left(\frac{b_0}{a}\right)^2`

Reviewer: ```suggestion
  :math:`k_{xy}=5.35 + 4\left(\frac{b_0}{a}\right)^2`.
```

>   :math:`k_{xy}=5.35 + 4\left(\frac{b_0}{a}\right)^2`

Reviewer: ```suggestion
.. math::
    k_{xy}=5.35 + 4\left(\frac{b_0}{a}\right)^2.
```

It is more convenient to use the shear force :math:`N_{xy}` instead of the shear

> stress component :math:`\tau_{xy}`. It leads to the relation:

Reviewer: ```suggestion
stress component :math:`\tau_{xy}`. It leads to the equation:
```

.. math::
   N_{xy}=q_1+q_c

where :math:`q_1` abd :math:`q_c` are the shear fluxes in the panel main surface
and its stiffener. They are given by:

.. math::
   q_1=\tau_{xy}t_s=2G_sh_0t_s\frac{\partial^2w}{\partial x\partial y}

> and

Reviewer: ```suggestion
and:
```

.. math::

>   q_c=\frac{G_ct_cp}{\ell}\left[h-2h_0+\frac{h_c}{2p}(f_1-f_2)\right]\frac{\partial^2w}{\partial x\partial y}

Reviewer: ```suggestion
  q_c=\frac{G_ct_cp}{\ell} \left(h - 2h_0 + \frac{h_c}{2p}(f_1-f_2)\right) \frac{\partial^2w}{\partial x\partial y}
```

where

> - :math:`G_s=\frac{E_s}{2(1+\nu_s)}` is the shear modulus of the panel main
>   surface;

Reviewer: ```suggestion
- :math:`G_s` is the shear modulus of the panel main
  surface:
.. math::
    G_s = \frac{E_s}{2(1 + \nu_s)};
```

- :math:`\frac{\partial^2w}{\partial x\partial y}` is the torsion strain of the
  panel main surface;

> - :math:`G_c=\frac{E_c}{2(1+\nu_c)}` is the shear coefficient of the stiffener;

Reviewer: ```suggestion
- :math:`G_c` is the shear coefficient of the stiffener:

.. math::
    G_c = \frac{E_c}{2(1 + \nu_c)};
```

- :math:`t_c` is the thickness of the stiffener;

- :math:`h_c` is the distance between the mean surfaces of the stiffener hat and
  foot;

> - :math:`h=h_c+\frac{t_c+t_s}{2}` is the distance between the mean surfaces of
>   the stiffener hat and the panel main surface;

Reviewer: Centering such an equation is easier to read:
```suggestion
- :math:`h` is the distance between the mean surfaces of
  the stiffener hat and the panel main surface:
  
.. math::
    h = h_c+\frac{t_c + t_s}{2};
```

- :math:`f_1` is the width of the foot of the stiffener;

- :math:`f_2` is the width of the hat of the stiffener;

- :math:`p` is the half-widht of the stiffener;

- :math:`R` is the radius of the circular part of the stiffener;

- :math:`\theta` is the angle of the circular part of the stiffener;

- :math:`\ell` is the length of the stiffener flank;

> - :math:`d=\frac{\ell-f_2}{2}-R\theta- :math:` is the half-lenght of the straight

Reviewer: I think that something wrong happened here:
![image](https://github.com/openturns/openturns/assets/31351465/e06ad013-ee29-4e11-945e-0fb455bc77a4)

Author: Right! Well done.

  part of the flank of the stiffener;

- :math:`A=\ell t_c` is the area of the section of an half-ondulation;

> - :math:`\bar{A}=A+pt_s+\frac{1}{2}(f_1-f_2)t_c` is the area of the section of
>   the full panel (main surface and stiffener) bounded by :math:`p`;

Reviewer: ```suggestion
- :math:`\bar{A}` is the area of the section of
  the full panel (main surface and stiffener) bounded by :math:`p`:
.. math::
    \bar{A} = A + pt_s + \frac{1}{2}(f_1 - f_2)t_c
```

> - :math:`h_0=\frac{1}{2\bar{A}}\left[A(h_c+t_c+t_s)+\frac{1}{2}t_c(f_1-f_2)(t_c+t_s)\right]`

Reviewer: Square brackets are usually used only with a particular meaning. Here, there is no specific interest to use them.
```suggestion
- :math:`h_0=\frac{1}{2\bar{A}} \left(A(h_c+t_c+t_s)+\frac{1}{2}t_c(f_1-f_2)(t_c+t_s)\right)`
```

Reviewer: And centering it will make it easier to read.

  is the distance between the mean surface of the panel main surface and the

>   global geometric center of the pa,nel;

Reviewer: ```suggestion
  global geometric center of the panel;
```

It leads to:

.. math::

>   N_{xy}=q_1(1+q_c/q1)=\tau_{xy}t_s\left[1+\frac{1}{4}\frac{G_ct_c}{G_st_s}\frac{\left(2p(h-2h_0)-h_c(f_1-f_2)\right)}{h_0\ell}\right]

Reviewer: ```suggestion
  N_{xy}=q_1(1+q_c/q1) = \tau_{xy}t_s \left(1 + \frac{1}{4}\frac{G_ct_c}{G_st_s} \frac{\left(2p(h-2h_0) - h_c(f_1-f_2)\right)}{h_0\ell}\right)
```

>   N_{xy}=q_1(1+q_c/q1)=\tau_{xy}t_s\left[1+\frac{1}{4}\frac{G_ct_c}{G_st_s}\frac{\left(2p(h-2h_0)-h_c(f_1-f_2)\right)}{h_0\ell}\right]

Reviewer: This the index "1":
```suggestion
  N_{xy}=q_1(1+q_c/q_1)=\tau_{xy}t_s\left[1+\frac{1}{4}\frac{G_ct_c}{G_st_s}\frac{\left(2p(h-2h_0)-h_c(f_1-f_2)\right)}{h_0\ell}\right]
```

and finally, :math:`(N_{xy})_{cr}` is given by:

.. math::

>   (N_{xy})_{cr}=\bigg[5.35 + 4\left(\frac{b_0}{a}\right)^2\bigg]\bigg[\frac{\pi^2}{b_0^2}\frac{E_st_s^3}{12(1-\nu_s^2)}\bigg]\bigg[1+\frac{1}{4}\frac{G_ct_c}{G_st_s}\frac{(2p(h-2h_0)-h_c(f_1-f_2))}{h_0\ell}\bigg]

Reviewer: ```suggestion
  (N_{xy})_{cr} = \left(5.35 + 4\left(\frac{b_0}{a}\right)^2\right) \left(\frac{\pi^2}{b_0^2}\frac{E_st_s^3}{12(1-\nu_s^2)}\right) \left(1+\frac{1}{4}\frac{G_ct_c}{G_st_s}\frac{(2p(h-2h_0)-h_c(f_1-f_2))}{h_0\ell}\right)
```

Author: I will keep \bigg and switch to parentheses, because I find it ugly when two blocks have delimiters of very different sizes.

Reviewer: The word "ugly" seems a little to much for this equation. Here is the output:

![image](https://github.com/openturns/openturns/assets/31351465/04175e0e-5343-4a3b-be3d-1195262e5657)

Notice the excess of parentheses in the third parenthesis, in the numerator of the fraction.

Reviewer: At this stage, I think that you are confusing two different goals in the presentation.
- You derive the equation for $N_{xy}$ and substitute for $k_{xy}$ and $D$, because that it is needed for alleviating a limitation of Exprtk that cannot handle exact derivation when there is a complicated evaluation tree.
- It is not necessary to get the full expanded expression to fully understand the model.

In the end, the expression:

![image](https://github.com/openturns/openturns/assets/31351465/aeab2b33-e9e8-4d99-b376-031a77030660)

is the simplest expression that describes the model isn'it?

Author: \bigg is a relief of a first version of the equations, where the front factor was expessed in terms of $r=\frac{b_0}{a}$. Now it is useless, well pointed!

For industrial constraints, the stiffener and the main surface are cut in the
same metal sheet, so :math:`E_c=E_s=E`, :math:`\nu_c=\nu_s=\nu`, :math:`t_c=t_s=t`.
The final expression of the critical shear force is then:

.. math::

>   (N_{xy})_{cr}=\bigg[5.35 + 4\left(\frac{b_0}{a}\right)^2\bigg]\bigg[\frac{\pi^2}{b_0^2}\frac{Et^3}{12(1-\nu^2)}\bigg]\bigg[1+\frac{1}{4}\frac{(2p(h-2h_0)-h_c(f_1-f_2))}{h_0\ell}\bigg]

Reviewer: ```suggestion
  (N_{xy})_{cr} = \left( 5.35 + 4\left(\frac{b_0}{a}\right)^2\right) \left(\frac{\pi^2}{b_0^2}\frac{Et^3}{12(1-\nu^2)}\right) \left(1+\frac{1}{4}\frac{(2p(h-2h_0)-h_c(f_1-f_2))}{h_0\ell}\right)
```

with:

- :math:`A=\ell t`

- :math:`\bar{A}=A+t\left(p+\frac{f_1-f_2}{2}\right)`

> - :math:`h_0=\frac{1}{2\bar{A}}\left[A(h_c+2t)+t^2(f_1-f_2)\right]`

Reviewer: ```suggestion
- :math:`h_0=\frac{A(h_c+2t)+t^2(f_1-f_2)}{2\bar{A}}`
```

> - :math:`A=\ell t`
> 
> - :math:`\bar{A}=A+t\left(p+\frac{f_1-f_2}{2}\right)`
> 
> - :math:`h_0=\frac{1}{2\bar{A}}\left[A(h_c+2t)+t^2(f_1-f_2)\right]`
> 
> - :math:`h=h_c+t`

Reviewer: Punctuation marks are needed here.
```suggestion
- :math:`A=\ell t`;

- :math:`\bar{A}=A+t\left(p+\frac{f_1-f_2}{2}\right)`;

- :math:`h_0=\frac{1}{2\bar{A}} \left(A(h_c+2t)+t^2(f_1-f_2)\right)`;

- :math:`h=h_c+t`.
```

> 

Reviewer: At this point, a classical ABC study defines a quantity of interest (QoI) e.g. the mean or the probability of exceeding a threshold. According to the example, that is the latter here. I suggest to introduce this topic here, independently of the the script.

Author: I don't really understand your point. Do you mean I have to write the description of a minimal study here?  There is no such description in other test-cases (viscous free fall, fire satellite, vertical deflection, wing weight and probably others) and I find it a good practice. It is up to the user to imagine an application, and the examples are here to give her/him ideas!

Reviewer: Please read the [use_case_cantilever_beam.html](http://openturns.github.io/openturns/latest/usecases/use_case_cantilever_beam.html). This example is often used as a reliability problem: a threshold is suggested.  There are other similar examples (e.g. [use_case_beam.html](http://openturns.github.io/openturns/latest/usecases/use_case_beam.html) and [use_case_flood_model.html](http://openturns.github.io/openturns/latest/usecases/use_case_flood_model.html)). Of course, the use case is not limited to the QoI that we suggest, but this gives a starting point for the user. You might suggest a similar idea in this particular study.

Author: IMO there is a confusion between the presentation of a use-case and the examples related to a given use-case. It is particularly clear in the http://openturns.github.io/openturns/latest/usecases/use_case_flood_model.html use-case, where the presentation of the use-case is followed by a study, which IMO is not the right place. Everything after "Moreover, we assume that the input random variables are independent." should be put in an example or two: probability of failure, sensitivity analysis, calibration, it is way too much to ask for the definition of a use-case.

Here are the use-cases without any reference to an UQ application:
  * [The Ishigami function](https://openturns.github.io/openturns/latest/usecases/use_case_ishigami.html)
  * [The logistic model](https://openturns.github.io/openturns/latest/usecases/use_case_logistic.html)
  * [The Ackley test case](https://openturns.github.io/openturns/latest/usecases/use_case_ackley.html)
  * [The Branin test case](https://openturns.github.io/openturns/latest/usecases/use_case_branin.html)
  * [The Chaboche mechanical model](https://openturns.github.io/openturns/latest/usecases/use_case_chaboche.html)
  * [Vertical deflection of a tube](https://openturns.github.io/openturns/latest/usecases/use_case_deflection_tube.html)
  * [A viscous free fall example](https://openturns.github.io/openturns/latest/usecases/use_case_viscous_fall.html)
  * [The Fire Satellite model](https://openturns.github.io/openturns/latest/usecases/use_case_fireSatellite.html)
  * [The Wing weight function](https://openturns.github.io/openturns/latest/usecases/use_case_wingweight.html)
  * [Coles datasets](https://openturns.github.io/openturns/latest/usecases/coles.html)
  * [The Linthurst data set](https://openturns.github.io/openturns/latest/usecases/use_case_linthurst.html)

so 11 over 15 use-cases.

BTW I have quite a lot of other tasks (including the review of your PRs) so I don't plan to spend much more time on this PR. We can discuss about the perimeter of what a use-case should be in a Technical Committee and then open an issue to fix either the 11 previous use-cases or the 4 others.

.. math::
    k_{xy} = 5.35 + 4 \left(\frac{b_0}{a}\right)^2.

It is more convenient to use the shear force :math:`N_{xy}` instead of the shear
stress component :math:`\tau_{xy}`. It leads to the equation:

.. math::
   N_{xy}=q_1+q_c

where :math:`q_1` abd :math:`q_c` are the shear fluxes in the panel main surface
and its stiffener. They are given by:

.. math::
   q_1=\tau_{xy}t_s=2G_sh_0t_s\frac{\partial^2w}{\partial x\partial y}

and:

.. math::
  q_c=\frac{G_ct_cp}{\ell} \left(h - 2h_0 + \frac{h_c}{2p}(f_1-f_2)\right) \frac{\partial^2w}{\partial x\partial y}

where:

- :math:`G_s` is the shear modulus of the panel main
  surface:

.. math::
    G_s = \frac{E_s}{2(1 + \nu_s)};

- :math:`\frac{\partial^2w}{\partial x\partial y}` is the torsion strain of the
  panel main surface;

- :math:`G_c` is the shear coefficient of the stiffener:

.. math::
    G_c = \frac{E_c}{2(1 + \nu_c)};

- :math:`t_c` is the thickness of the stiffener;

- :math:`h_c` is the distance between the mean surfaces of the stiffener hat and
  foot;

- :math:`h` is the distance between the mean surfaces of
  the stiffener hat and the panel main surface:

.. math::

>     h = h_c+\frac{t_c + t_s}{2}

Reviewer: ```suggestion
    h = h_c+\frac{t_c + t_s}{2};
```

- :math:`f_1` is the width of the foot of the stiffener;

- :math:`f_2` is the width of the hat of the stiffener;

- :math:`p` is the half-widht of the stiffener;

- :math:`R` is the radius of the circular part of the stiffener;

- :math:`\theta` is the angle of the circular part of the stiffener;

- :math:`\ell` is the length of the stiffener flank;

- :math:`d=\frac{\ell-f_2}{2}-R\theta` is the half-lenght of the straight
  part of the side of the stiffener;

- :math:`A=\ell t_c` is the area of the section of an half-ondulation;

- :math:`\bar{A}` is the area of the section of
  the full panel (main surface and stiffener) bounded by :math:`p`:

.. math::
    \bar{A} = A + pt_s + \frac{1}{2}(f_1 - f_2)t_c

> - :math:`h_0=\frac{1}{2\bar{A}} \left(A(h_c+t_c+t_s)+\frac{1}{2}t_c(f_1-f_2)(t_c+t_s)\right)`
>   is the distance between the mean surface of the panel main surface and the
>   global geometric center of the panel;

Reviewer: The equation needs to be centered and the sentence must finish with a final point.
```suggestion
- :math:`h_0` is the distance between the mean surface of the panel main surface and the
  global geometric center of the panel:

.. math::
    h_0 = \frac{1}{2\bar{A}} \left(A(h_c+t_c+t_s)+\frac{1}{2}t_c(f_1-f_2)(t_c+t_s)\right).
```

It leads to:

.. math::

>   N_{xy}=q_1(1+q_c/q_1) = \tau_{xy}t_s \left(1 + \frac{1}{4}\frac{G_ct_c}{G_st_s} \frac{\left(2p(h-2h_0) - h_c(f_1-f_2)\right)}{h_0\ell}\right)

Reviewer: ![image](https://github.com/openturns/openturns/assets/31351465/5bc79561-1ac2-4981-95d6-302084173c37)

```suggestion
  N_{xy}=q_1(1+q_c/q_1) = \tau_{xy}t_s \left(1 + \frac{1}{4}\frac{G_ct_c}{G_st_s} \frac{2p(h-2h_0) - h_c(f_1-f_2)}{h_0\ell}\right)
```
